os/adding-swap.md

```diff
@@ -0,0 +1,78 @@
+# Adding swap in CoreOS/
+Swap is the process of moving pages of memory to a designated part of the hard disk, freeing up space when needed. Swap can be used to alleviate problems with low-memory environments.
+
+By default CoreOS does not include a partition for swap, however one can configure their system to have swap.
+
+## Manually adding a swapfile on ext4
+
+Run these commands **as root** to create a 512 megabyte swapfile.
+
+```
+$ fallocate -l 512m /var/512MiB.swap
+$ chmod 600 /var/512MiB.swap
+$ chown root /var/
+$ mkswap /var/512MiB.swap
+```
+
+**Note**: The partition on which you are creating the swapfile must meet the following requirements:
+
+* Must not be of type **tmpfs**, **btrfs**, or **xfs**.
+* Must be large enough for your swapfile size.
+
+To verify that our swapfile meets those requirements, run the following command:
+
+```
+$ df -Th
+Filesystem     Type      Size  Used Avail Use% Mounted on
+[...]
+/dev/sdXN      ext4      2.0G  3.0M  1.8G   1% /var
+```
+
+The block device `/dev/sdXN` the correct filesystem type and has enough space for a 512MiB swapfile, so we can proceed.
+
+### Create a unit to initialize the swap on boot
+
+Create a unit file at `/var/units/swapon.service` with these contents:
```

Comment:
@ElijahCaine This should be based instead on this approach: https://github.com/coreos/bugs/issues/429#issuecomment-153979506

That is, wrap all this procedure in a systemd unit file (you get up-to-date paths for free!). You can test the unit file on any running coreos instance pretty easily.